TestFoundation/TestHTTPCookie.swift

```diff
@@ -171,6 +173,35 @@ class TestHTTPCookie: XCTestCase {
         XCTAssertEqual(cookies[0].path, "/")
     }
     
+    func test_cookiesWithExpiresAsLastAttribute() throws {
+        let cookies = HTTPCookie.cookies(withResponseHeaderFields: [
+            "Set-Cookie": "AAA=111; path=/; domain=.example.com; expires=Sun, 16-Aug-2025 22:39:54 GMT, BBB=222; path=/; domain=.example.com; HttpOnly; expires=Sat, 15-Feb-2014 22:39:54 GMT"
+        ], for: try URL(string: "http://www.example.com/").unwrapped())
+        XCTAssertEqual(cookies.count, 2)
+        XCTAssertEqual(cookies[0].name, "AAA")
+        XCTAssertEqual(cookies[1].name, "BBB")
+    }
+
+    func test_cookiesWithResponseHeaderTrimsNames() throws {
+        do {
+            let cookies = HTTPCookie.cookies(withResponseHeaderFields: [
+                "Set-Cookie": "AAA =1; path=/; domain=.example.com; expires=Sun, 16-Aug-2025 22:39:54 GMT, BBB=2; path=/; domain=.example.com; HttpOnly; expires=Sat, 15-Feb-2014 22:39:54 GMT"
+            ], for: try URL(string: "http://www.example.com/").unwrapped())
+            XCTAssertEqual(cookies.count, 2)
+            XCTAssertEqual(cookies[0].name, "AAA")
+            XCTAssertEqual(cookies[1].name, "BBB")
+        }
+
+        do {
+            let cookies = HTTPCookie.cookies(withResponseHeaderFields: [
+                "Set-Cookie": " AAA=1; path=/; domain=.example.com; expires=Sun, 16-Aug-2025 22:39:54 GMT, BBB=2; path=/; domain=.example.com; HttpOnly; expires=Sat, 15-Feb-2014 22:39:54 GMT"
+            ], for: try URL(string: "http://www.example.com/").unwrapped())
+            XCTAssertEqual(cookies.count, 2)
+            XCTAssertEqual(cookies[0].name, "AAA")
+            XCTAssertEqual(cookies[1].name, "BBB")
+        }
+    }
+
```

Comment:
Probably worth adding in some broken cookies and ones where the value part has `=` in it to test the parsing. I would also add in `XCTAssertEqual(cookies[0].value...` checks to validate the values.